File: killrvideo-service-videocatalog/src/main/java/com/killrvideo/service/video/request/GetLatestVideoPreviewsRequestData.java
package com.killrvideo.service.video.request;

import com.killrvideo.dse.dto.CustomPagingState;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.Instant;
import java.util.Optional;
import java.util.UUID;

@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
@Getter @AllArgsConstructor
public class GetLatestVideoPreviewsRequestData {
    private final CustomPagingState pageState;
    private final int pageSize;
    private final Optional<Instant> startDate;
    private final Optional<UUID> startVideoId;
}

File: killrvideo-service-videocatalog/src/main/kotlin/com/killrvideo/service/video/request/GetLatestVideoPreviewsRequestData.kt
package com.killrvideo.service.video.request

import com.killrvideo.dse.dto.CustomPagingState
import java.time.Instant
import java.util.*

data class GetLatestVideoPreviewsRequestData(
    val pageState: CustomPagingState,
    val pageSize: Int,
    val startDate: Instant?,
    val startVideoId: UUID?
)
